extensions/iceberg/src/main/java/io/deephaven/iceberg/base/IcebergUtils.java

```diff
@@ -105,17 +105,56 @@ public static io.deephaven.qst.type.Type<?> convertToDHType(@NotNull final Type
             case FIXED: // Fall through
             case BINARY:
                 return io.deephaven.qst.type.Type.find(byte[].class);
+            case LIST:
+                return convertToDHArrayType(icebergType.asListType());
             case UUID: // Fall through
             case STRUCT: // Fall through
+            case MAP: // Fall through
+            default:
+                // TODO (DH-19247) Add support for more types like UUID, Struct, Map
+                throw new TableDataException("Iceberg columns of type " + typeId.name() + "is not supported when" +
+                        " reading data from Iceberg tables");
+        }
+    }
+
+    /**
+     * Convert an Iceberg data type to a Deephaven type. Used for reading data from Iceberg tables.
+     *
+     * @param icebergListType The Iceberg list type to be converted.
+     * @return The converted Deephaven type.
+     */
+    private static io.deephaven.qst.type.Type<?> convertToDHArrayType(@NotNull final Types.ListType icebergListType) {
+        final Type.TypeID elementTypeId = icebergListType.elementType().typeId();
+        switch (elementTypeId) {
+            case BOOLEAN:
+            case DOUBLE:
+            case FLOAT:
+            case INTEGER:
+            case LONG:
+            case STRING:
+            case TIMESTAMP:
+            case DATE:
+            case TIME:
+            case DECIMAL:
+                try {
+                    return convertToDHType(icebergListType.elementType()).arrayType();
+                } catch (final TableDataException e) {
```

Comment:
I'm wondering if we might want to use the DbVector types here instead of native array.